src/main.ml

```diff
@@ -57,32 +84,69 @@ let check_npm_deps cli =
     OpamCoreConfig.update ~verbose_level:0 ();
     OpamGlobalState.with_ `Lock_none @@ fun gt ->
     OpamSwitchState.with_ `Lock_write gt @@ fun st ->
-    let () =
-      match match_version_against_constraint "1.0.1" "^1.0.0" with
-      | true -> print_endline "TRUE"
-      | false -> print_endline "FALSE"
-    in
     let npm_depexts =
       List.filter_map
         (fun pkg ->
           let depexts = depexts pkg st.opams in
-          match OpamSysPkg.Set.cardinal depexts with
-          | 0 -> None
-          | _ -> Some (pkg, depexts))
+          match depexts with [] -> None | _ -> Some (pkg, depexts))
         OpamPackage.Set.(elements @@ st.installed)
     in
     let () =
       match npm_depexts with
       | [] -> ()
       | l ->
-          print_endline "Found the following npm dependencies in opam files:";
-          print_endline
-            (OpamStd.List.concat_map " "
-               (fun (pkg, npm_deps) ->
-                 Printf.sprintf "pkg: %s, depexts: %s\n"
-                   (OpamPackage.to_string pkg)
-                   (OpamSysPkg.Set.to_string npm_deps))
-               l)
+          List.iter
+            (fun (opam_pkg, npm_pkgs_and_constraints) ->
+              List.iter
+                (fun (npm_pkgs, npm_constraint) ->
+                  OpamSysPkg.Set.iter
+                    (fun npm_pkg ->
+                      let open EsyLib in
+                      let path =
+                        Path.(
+                          currentPath () / "node_modules"
+                          / OpamSysPkg.to_string npm_pkg
+                          / "package.json")
+                      in
+                      let installed_version =
+                        try Ok (RunAsync.runExn (Of_package_json.read path))
+                        with _exn -> Error ()
+                      in
+                      match installed_version with
+                      | Error () ->
+                          print_endline
+                            (Printf.sprintf
+                               "Error: opam package \"%s\" requires npm package \"%s\" \
```

Comment:
I wonder if the command should return an error code when some error has been found?

Reply:
I can see a flag with an option, but for now it's perfect to just print it IMO

Reply:
I meant the opam plugin binary status code returned after the process ends.

Reply:
E.g. if no errors are found, we'd return 0, but if some error is found, we can return 1.

Reply:
Added a flag `--dry-run` in https://github.com/jchavarri/opam-check-npm-deps/pull/2/commits/c14926612423f0b6e53b69c3b983e1355ae07558.